lib/declare_schema/model/field_spec.rb

```diff
@@ -166,28 +162,33 @@ def same_attributes?(col_spec)
         end
       end
 
-      def same_charset_and_collation?(col_spec)
-        if type.in?([:text, :string])
-          current_settings = collation_and_charset_for_column(col_spec)
-          current_settings[:collation] == collation
-        else
-          true
-        end
+      def same_charset_and_collation?(table_name, col_spec)
+        current_collation_and_charset = collation_and_charset_for_column(table_name, col_spec)
```

Comment:
No need for the above variable if the type isn't right, right? Assuming so, can you move this local variable and the part after the `||` below into another private method? That will keep the logic really clean.

Reply:
Good call. I went ahead and move the `type` check up and out of this method as it's out of scope for what this method's responsibilities are.